scripts/sync-locale.ts

```diff
@@ -0,0 +1,243 @@
+import { readFileSync, writeFileSync } from 'node:fs'
+import { basename, resolve } from 'node:path'
+import glob from 'fast-glob'
+import consola from 'consola'
+import { isArray, isObject, isString } from 'lodash-unified'
+import { localeRoot, normalizePath } from '@element-plus/build-utils'
+
+interface LocaleObject {
+  [key: string]: LocaleObject | string | Array<string>
+}
+
+const langRoot = resolve(localeRoot, 'lang')
+const enFile = normalizePath(resolve(langRoot, 'en.ts'))
+const localePath = normalizePath(resolve(`${langRoot}/!(en).ts`))
+const commentInfo = '// to be translated'
+
+function parseLocaleFile(content: string): LocaleObject {
+  try {
+    const untranslatedRexExp = new RegExp(
```

Comment:
This regular expression can be put outside, without having to create it every time the function is called.